1-Draft/RFCNNNN-Side-By-Side-Support.md

```diff
@@ -0,0 +1,253 @@
+-------                    ---------------
+6.2.1                      C:\ProgramFiles\PowerShell\6.2.1
+7.0.100-preview1           C:\ProgramFiles\PowerShell\7.0.100-preview1
+```
+
+### Launching a specific version of PowerShell
+
+```bash
+# This launches the latest version of PowerShell, gets the installation
+# location of the specified version, launches pwsh for that version, and exits
+# when the launched version exits
+pwsh -Version 6.2.1
+# This does the same thing
+pwsh -v 6.2.1
+```
+
+```output
+# A new window with the desired version of PowerShell open.
+# If it is possible to do have this window on a new tab in the new Terminal,
+# that would be great.
+```
+
+## Specification
+
+* create a `Get-InstalledPSVersion [<CommonParameters>]` cmdlet that returns
+the installed versions of PowerShell along with their locations
+* update the `-version` argument of `pwsh.exe` to attempt to launch a specific
+version of PowerShell
+* add a `-ListVersions` argument to `pwsh.exe` that internally invokes
+`Get-InstalledPSVersion`, formats and outputs the results, and exits
+* update the release packaging process to produce versioned packages
+
+## Alternate Proposals and Considerations
+
+### Include Windows PowerShell version support
+
+It would be worth considering having `Get-InstalledPSVersion` return the
+installed version of Windows PowerShell along with its location, and having
+`pwsh -version` support launching Windows PowerShell. That would allow users to
+launch any installed PowerShell version from `pwsh.exe`, making it easy to set
+up automation with one command to invoke. Arguments from `pwsh.exe` could be
+mapped to arguments on `powershell.exe`.
+
+### Are containers and Docker enough to solve this need?
+
+While discussing this with a colleague of mine, they asked if this is really
+needed, or if containers and Docker solve this problem already. While I believe
+containers can allow users to use different versions of PowerShell, I also
+suspect that most users don't know how to set that up, if they even use Docker
+at all. I think it would be easier for everyone if we could just launch the
+proper version of PowerShell via `pwsh.exe`.
+
+### Maintain backwards compatibility with pwsh.exe -version
+
+In PowerShell Core 6.2, if you invoke `pwsh -version` you get the current
+PowerShell version. That support could remain in place if this RFC is approved,
+returning the current (latest) version when invoked without a version, or
+launching the requested version when invoked with a version.
+
+### Windows Update
+
+If PowerShell is added back as a component of Windows, some consideration needs
+to be given to Windows Update. With side-by-side installations, it would not
+make much sense to update older versions, so the most recently installed stable
+version should be treated as the default version to check for updates.
+
+Unfortunately I do not have much insight into what Windows Update requires.
```

Comment:
Windows Update requires that the path to the binaries don't change.  This means that we don't have even a complete version as the folder name hence it's just `6` and not `6.1` or `6.2`.  This means that this may limit the usefulness to just major versions.